beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -891,7 +918,7 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
                 .network_globals()
                 .peers
                 .read()
-                .synced_peers()
+                .synced_peers_for_epoch(batch_id, &self.peers)
```

Comment:
Instead, you can iterate `self.peers` here and filter by peers that have epoch `batch_id`. All peers in the batch are synced connected peers

Reply:
Ohh yeah good point, i'll make the changes.